src/tests/ftest/util/dmg_utils.py

```diff
@@ -1340,9 +1340,9 @@ def get_json_response(data, key, description):
     """
     response = {}
     try:
-        if data['error']:
+        if 'error' in data and data['error']:
             raise CommandFailure("{} failed: {}".format(description, data['error']))
-        if len(data['response']['host_errors']) > 0:
+        if 'response' in data and len(data['response']['host_errors']) > 0:
```

Comment:
Why is this necessary? If the json result doesn't contain `error` or `response`, isn't that a problem with the json returned by the `dmg` command? I.e. even if the command fails, the json should contain at least `error`

Reply:
Done. the changes based on @phender and @daltonbohning .